app/questionnaire/questionnaire_schema.py

```diff
@@ -169,3 +140,55 @@ def get_repeat_rule(group):
     def get_questions_for_block(block_json):
         for question_json in block_json.get('questions', []):
             yield question_json
+
+    def _parse_schema(self):
+        self._sections_by_id = self._get_sections_by_id()
+        self._groups_by_id = get_nested_schema_objects(self._sections_by_id, 'groups')
+        self._blocks_by_id = get_nested_schema_objects(self._groups_by_id, 'blocks')
+        self._questions_by_id = get_nested_schema_objects(self._blocks_by_id, 'questions')
+        self._answers_by_id = get_nested_schema_objects(self._questions_by_id, 'answers')
+        self.error_messages = self._get_error_messages()
+        self.aliases = self._get_aliases()
+
+    def _get_sections_by_id(self):
+        return collections.OrderedDict(
+            (section['id'], section)
+            for section in self.json.get('sections', [])
+        )
+
+    def _get_error_messages(self):
+        messages = error_messages.copy()
+        if 'messages' in self.json:
+            messages.update(self.json['messages'])
+
+        return messages
+
+    def _get_aliases(self):
+        aliases = {}
+        for question in self.questions:
+            for answer in question.get('answers', []):
+                if 'alias' in answer:
+                    if answer['alias'] in aliases:
+                        raise Exception(
+                            'Duplicate alias found: ' + answer['alias'])
+                    aliases[answer['alias']] = {
+                        'answer_id': answer['id'],
+                        'repeats': answer['type'] == 'Checkbox' or question['type'] == 'RepeatingAnswer',
+                    }
+
+        return aliases
+
+
+def get_nested_schema_objects(parent_object, list_key, id_field='id'):
```

Comment:
do we need `id_field` if we never override it?

Reply:
Fair point - will remove.